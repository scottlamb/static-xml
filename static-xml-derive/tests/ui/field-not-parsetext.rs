// Copyright (C) 2022 Scott Lamb <slamb@slamb.org>
// SPDX-License-Identifier: MIT OR Apache-2.0

use static_xml_derive::Deserialize;

struct Foo;

#[derive(Deserialize)]
struct Outer {
    foo: Foo,
}

fn main() {}
