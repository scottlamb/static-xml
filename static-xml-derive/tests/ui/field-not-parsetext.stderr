error[E0277]: the trait bound `Foo: ParseText` is not satisfied
  --> tests/ui/field-not-parsetext.rs:10:5
   |
8  | #[derive(Deserialize)]
   |          ----------- this tail expression is of type `&mut Foo`
9  | struct Outer {
10 |     foo: Foo,
   |     ^^^^^^^^ the trait `ParseText` is not implemented for `Foo`
   |
   = help: the following other types implement trait `ParseText`:
             String
             bool
             f32
             f64
             i128
             i16
             i32
             i64
           and $N others
   = note: required for `Foo` to implement `Deserialize`
   = note: required for `Foo` to implement `DeserializeElementField`

error[E0277]: the trait bound `Foo: ParseText` is not satisfied
  --> tests/ui/field-not-parsetext.rs:10:5
   |
10 |     foo: Foo,
   |     ^^^^^^^^ the trait `ParseText` is not implemented for `Foo`
   |
   = help: the following other types implement trait `ParseText`:
             String
             bool
             f32
             f64
             i128
             i16
             i32
             i64
           and $N others
   = note: required for `Foo` to implement `Deserialize`
   = note: required for `Foo` to implement `DeserializeElementField`
